OrbitQt/orbittreeview.cpp

```diff
@@ -47,122 +44,115 @@ OrbitTreeView::OrbitTreeView(QWidget* parent)
   connect(this, SIGNAL(clicked(const QModelIndex)), this,
           SLOT(OnClicked(const QModelIndex)));
 
-  connect(this->verticalScrollBar(), SIGNAL(rangeChanged(int, int)), this,
+  connect(verticalScrollBar(), SIGNAL(rangeChanged(int, int)), this,
           SLOT(OnRangeChanged(int, int)));
 }
 
 //-----------------------------------------------------------------------------
-OrbitTreeView::~OrbitTreeView() {
-  delete m_Model;
-  delete m_Timer;
-}
-
-//-----------------------------------------------------------------------------
-void OrbitTreeView::Initialize(DataViewType a_Type) {
-  m_Model = new OrbitTableModel(a_Type);
-  setModel(m_Model);
+void OrbitTreeView::Initialize(DataViewType type) {
+  model_ = std::make_unique<OrbitTableModel>(type);
+  setModel(model_.get());
   header()->resizeSections(QHeaderView::ResizeToContents);
 
-  if (!m_Model->IsSortingAllowed()) {
-    // Don't do setSortingEnabled(m_Model->IsSortingAllowed()); as with true it
+  if (!model_->IsSortingAllowed()) {
+    // Don't do setSortingEnabled(model_->IsSortingAllowed()); as with true it
     // forces a sort by the first column.
     setSortingEnabled(false);
   } else {
     std::pair<int, Qt::SortOrder> column_and_order =
-        m_Model->GetDefaultSortingColumnAndOrder();
+        model_->GetDefaultSortingColumnAndOrder();
     sortByColumn(column_and_order.first, column_and_order.second);
   }
 
-  if (m_Model->GetUpdatePeriodMs() > 0) {
-    m_Timer = new QTimer(this);
-    connect(m_Timer, SIGNAL(timeout()), this, SLOT(OnTimer()));
-    m_Timer->start(m_Model->GetUpdatePeriodMs());
+  if (model_->GetUpdatePeriodMs() > 0) {
+    timer_ = std::make_unique<QTimer>(this);
```

Comment:
Don't pass `this` as a parent here. Qt's QObject-parent-system is a resource management system.
So when a QObject gets deconstructed it deletes all its children. Since the resource is managed here
by std::unique_ptr you shouldn't delegate its management to the tree-view class. In unfortunate situation
this leads to double-deletions. This won't happen here because the unique_ptr deconstructs the timer before
the treeview gets deconstructed. But if the timer were moved somewhere else this would be a problem.

Reply:
removed this.